src/test/java/se/sundsvall/contactsettings/service/util/FilterEvaluationUtilsTest.java

```diff
@@ -0,0 +1,255 @@
+						DelegateFilterRule.create().withAttributeName("key2").withOperator("NOT_EQUALS").withAttributeValue("value4"),
+						DelegateFilterRule.create().withAttributeName("key2").withOperator("NOT_EQUALS").withAttributeValue("value-no-match")))), true),
+
+			Arguments.of(
+				"Match with mixed EQUALS and NOT_EQUALS-rules in multiple filters",
+				Map.of(
+					"key1", List.of("value1", "value2"),
+					"key2", List.of("value3", "value4", "value5")),
+				List.of(
+					DelegateFilterEntity.create().withFilterRules(List.of(
+						DelegateFilterRule.create().withAttributeName("key1").withOperator("EQUALS").withAttributeValue("value2"),
+						DelegateFilterRule.create().withAttributeName("key2").withOperator("NOT_EQUALS").withAttributeValue("value-no-match"))),
+					DelegateFilterEntity.create().withFilterRules(List.of(
+						DelegateFilterRule.create().withAttributeName("key1").withOperator("NOT_EQUALS").withAttributeValue("value-no-match"),
+						DelegateFilterRule.create().withAttributeName("key2").withOperator("EQUALS").withAttributeValue("value4")))), true),
+
+			Arguments.of(
+				"Match with mixed EQUALS and NOT_EQUALS-rules in multiple filters, where one rule-condition in one filter is not fulfilled (OR-condition)",
+				Map.of(
+					"key1", List.of("value1", "value2"),
+					"key2", List.of("value3", "value4", "value5")),
+				List.of(
+					DelegateFilterEntity.create().withFilterRules(List.of(
+						DelegateFilterRule.create().withAttributeName("key1").withOperator("EQUALS").withAttributeValue("value2"),
+						DelegateFilterRule.create().withAttributeName("key2").withOperator("NOT_EQUALS").withAttributeValue("value-no-match"))),
+					DelegateFilterEntity.create().withFilterRules(List.of(
+						DelegateFilterRule.create().withAttributeName("key1").withOperator("NOT_EQUALS").withAttributeValue("value-no-match"),
+						DelegateFilterRule.create().withAttributeName("key2").withOperator("EQUALS").withAttributeValue("value-no-match")))), true),
+
+			Arguments.of(
+				"No match with mixed EQUALS and NOT_EQUALS-rules in multiple filters.",
+				Map.of(
+					"key1", List.of("value1", "value2"),
+					"key2", List.of("value3", "value4", "value5")),
+				List.of(
+					DelegateFilterEntity.create().withFilterRules(List.of(
+						DelegateFilterRule.create().withAttributeName("key1").withOperator("EQUALS").withAttributeValue("value-no-match"),
+						DelegateFilterRule.create().withAttributeName("key2").withOperator("NOT_EQUALS").withAttributeValue("value-no-match"))),
+					DelegateFilterEntity.create().withFilterRules(List.of(
+						DelegateFilterRule.create().withAttributeName("key1").withOperator("NOT_EQUALS").withAttributeValue("value-no-match"),
+						DelegateFilterRule.create().withAttributeName("key2").withOperator("EQUALS").withAttributeValue("value-no-match")))), false),
+
+			Arguments.of(
+				"Match if input query is empty",
```

Comment:
This should eveluate to false right? For the rule to pass the keys of DelegateFilterEntity should be a subset of inputFilter. Since key1 and key2 is present in both there should not be any match

Reply:
Fixed